### ISSUE TITLE
Keychain plugin fails with .local domain name.

### ISSUE THREAD

**Issue by matthewhembree** [ohmyzsh/ohmyzsh#8380]
Keychain seems to write the environment files with the fqdn (which includes .local, in my case).

The Keychain plugin sources $HOME/.keychain/$SHORT_HOST-sh, which is not found. This leaves the SSH_AUTH_SOCK as the default macos one set by launchd. Identities never get added to the Keychain instantiated agent.

ssh-add -L (keychain -L)
"The agent has no identities."

I have added a line to my .zshrc as a workaround:
source "$HOME/.keychain/$HOST-sh"

macos 10.15.1

**Comment by larson-carter** [comment]
Hello @matthewhembree Can you give me some more details on your enviroment? Such as DNS IP-Ranges DHCP?

**Comment by mcornella** [comment]
You're right, as pointed out in [this commit comment](https://github.com/ohmyzsh/ohmyzsh/commit/95a3b2768033627d9ba3e347f368dffae92906b7#commitcomment-32794010) they use `$HOSTNAME` instead:

https://github.com/funtoo/keychain/blob/7e37e4b63529f4a1caae9ca5e6db42b10a6fd2f3/keychain#L1373-L1375

This is a simple change, can you submit a PR?

**Comment by matthewhembree** [comment]
I'm not sure if it's relevant to go down the troubleshooting path. I'll compare may work with my home environment later.

```$HOSTNAME``` is empty on my system. ```uname -n``` returns the ```.local``` suffix.

It seems that the plugin could just call keychain as below and avoid all ambiguity when sourcing the pidfile:
```sh
keychain ${^options:-} --agents ${agents:-gpg} ${^identities} --host $SHORT_HOST
```
I'm not sure what the [NFS compatibility](https://github.com/funtoo/keychain/blob/7e37e4b63529f4a1caae9ca5e6db42b10a6fd2f3/keychain#L1368) issue would be.

**Comment by mcornella** [comment]
Right, [`$hostopt` is set when passing `--host`](https://github.com/funtoo/keychain/blob/7e37e4b63529f4a1caae9ca5e6db42b10a6fd2f3/keychain#L1271-L1274). That looks like a cleaner solution.

I'm not sure if we need to think through the behavior when upgrading the plugin. What would happen if keychain is running and the plugin is upgraded? If not on a `.local` domain, `$SHORT_HOST` and the keychain method are equivalent, so there wouldn't be a problem? I don't have a Mac to test this.

**Comment by matthewhembree** [comment]
My understanding is keychain is just a helper to set the envars and launch the agents, if necessary. So, very short lifetime.

On upgrade, my work Mac would start working. There would just be additional files in ```~/..keychain```; same ```$SSH_AGENT_PID``` and ```$SSH_AUTH_SOCK```. My home Mac would still work.

You make an interesting point about upgrading. It might be useful to have a migrate.sh file be called after upgrade.sh. It's outside of the scope of this issue and perhaps really hard to maintain if skipping commits when pulling.

**Comment by mcornella** [comment]
Ok let's ship your solution then. Can you submit a PR?

**Comment by mcornella** [comment]
Closed by #8436

**Comment by elnull** [comment]
It's better to rewrite host argument ```--host ${SHORT_HOST:-$HOST}``` to fallback to ```HOST``` when ```SHORT_HOST``` is not defined.

**Comment by matthewhembree** [comment]
I understand the need for fallbacks, but `SHORT_HOST` should always be [defined](https://github.com/ohmyzsh/ohmyzsh/blob/ea3e666e04bfae31b37ef42dfe54801484341e46/oh-my-zsh.sh#L51-L57) at OMZ [init](https://github.com/ohmyzsh/ohmyzsh/wiki/Design#variables-omz-provides-or-modifies).

It's unlikely to become `unset` before the plugin loads. Am I missing a possibility?

**Comment by elnull** [comment]
I use this plugin with antibody plugin manager without OMZ.

**Comment by akadaedalus** [comment]
SHORT_HOST is also unavailable in my system.  It may have to do with using antigen as a package manager, though it does use OMZ init as well.  I had to set the value in my zshrc.

**Comment by mcornella** [comment]
Fixed in 00f3fa2c40c707c84d3dc416a1e1db25d78fcfaf.
